### PR TITLE
[INLONG-9297][Manager] Support configuring multiple sink types of tasks under a single stream

### LINKED ISSUE #9297: [Feature][Manager] Support configuring multiple sink types of tasks under a single stream
### Description

Support configuring multiple sink types of tasks under a single stream.
Currently, only one type of sink task is allowed to be configured in a stream, such as hive and iceberg.
Not meeting the requirement to write data from the same mq to multiple types of database and tables.
Therefore, the relevant configuration information for job start and stop is moved from the group level to the stream level to meet the requirement of writing data from mq to various database tables.

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by fuweng11**
### Prepare a Pull Request

- Fixes #9297

### Motivation

 Support configuring multiple sink types of tasks under a single stream.

### Modifications

 Support configuring multiple sink types of tasks under a single stream.

### CODE REVIEW

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/SinkType.java [hunk 1]
```diff
@@ -39,4 +45,23 @@ public class SinkType extends StreamType {
      * Details: <a href="https://www.tencentcloud.com/products/cls">CLS</a>
      */
     public static final String CLS = "CLS";
+
+    public static final Set<String> SORT_FLINK_SINK = new HashSet<>();
+    public static final Set<String> SORT_STANDALONE_SINK = new HashSet<>();
+
+    static {
+        SORT_FLINK_SINK.addAll(
```
Comment (healchow) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/SinkType.java:53]: This method is not very flexible. If a new type is added, the code here needs to be updated. Can we achieve this functionality through annotations?

Comment (fuweng11) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/SinkType.java]: Done

**Review by vernedeng — APPROVED** [review]

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/SinkType.java [hunk 2]
```diff
@@ -39,4 +47,27 @@ public class SinkType extends StreamType {
      * Details: <a href="https://www.tencentcloud.com/products/cls">CLS</a>
      */
     public static final String CLS = "CLS";
+
+    public static final Set<String> SORT_FLINK_SINK = new HashSet<>();
+
+    public static final Set<String> SORT_STANDALONE_SINK = new HashSet<>();
+
+    public static boolean containSortFlinkSink(List<String> sinkTypes) {
```
Comment (vernedeng) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/SinkType.java:55]: use  Set::retainAll to get the intersection of two sets

Comment (fuweng11) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/SinkType.java:55]: Done.

**Review by vernedeng — DISMISSED** [review]

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/SinkType.java [hunk 3]
```diff
@@ -17,9 +17,17 @@
...
+import org.springframework.stereotype.Component;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
 /**
  * Constants of sink type.
  */
+@Component
```
Comment (healchow) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/SinkType.java:30]: I mean, you can customize two annotations: one for types supporting `Sort, such as `@SupportSort`, and another for types supporting `SortStandalone`, such as `@SupportSortStandalone`.

On each Sink type, specify the supported annotation types. During the InLong-Manager startup, use reflection to read specific annotations and categorize Sink types into different `Set`.

With this approach, when adding a new Sink type, you only need to annotate it with one or both of the mentioned annotations, eliminating the need to maintain a configuration file.

**Review by luchunliang — DISMISSED** [review]

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/SupportSortType.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@Target({ElementType.FIELD})
+public @interface SupportSortType {
```
Comment (healchow) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/SupportSortType.java:29]: Can you please add some javadoc for this annotation?

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/SortType.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.manager.common.consts;
+
+public enum SortType {
```
Comment (healchow) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/SortType.java:20]: Javadoc, please.

Comment (fuweng11) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/SortType.java:23]: Done.

**Review by luchunliang — APPROVED** [review]

**Review by healchow — APPROVED** [review]
LGTM

**Review by luchunliang — APPROVED** [review]

**Review by vernedeng — APPROVED** [review]

**Pull request merged**